contracts/votingMachines/AbsoluteVote.sol

```diff
@@ -45,44 +45,59 @@ contract AbsoluteVote is IntVoteInterface {
    * @dev Check that the proposal is votable (open and not executed yet)
    */
     modifier votable(bytes32 _proposalId) {
-        require(proposals[_proposalId].open);
+        require(proposals[_proposalId].open, "proposal is not votable");
         _;
     }
 
+    /**
+    * @dev initialize
+    * @param _precReq requre percentage for absolure majority
+    * @param _voteOnBehalf enable vote on behalf
+    * @param _organization organization
+    * @param _callbacks should fulfill voting callbacks interface.
+    * @param _authorizedToPropose only this address allow to propose (unless it is zero)
+    */
+    function initialize(uint256 _precReq,
+                        address _voteOnBehalf,
+                        address _organization,
+                        address _callbacks,
+                        address _authorizedToPropose)
+    external
+    initializer {
+        require(_precReq <= 100 && _precReq > 0, "wrong precReq");
+        parameters = Parameters({
```

Comment:
Why not save them directly as variables of the contract instead of keeping the struct? (In all the contracts)

Reply:
it looks better in struct . though need to check the gas diff for this 2 options.